pybamm/expression_tree/symbol.py

```diff
@@ -360,79 +360,109 @@ def __repr__(self):
 
     def __rsub__(self, other):
         """return a :class:`Subtraction` object"""
-        return pybamm.Subtraction(other, self)
+        return pybamm.simplify_if_constant(
+            pybamm.Subtraction(other, self), keep_domains=True
+        )
 
     def __mul__(self, other):
         """return a :class:`Multiplication` object"""
-        return pybamm.Multiplication(self, other)
+        return pybamm.simplify_if_constant(
+            pybamm.Multiplication(self, other), keep_domains=True
+        )
 
     def __rmul__(self, other):
         """return a :class:`Multiplication` object"""
-        return pybamm.Multiplication(other, self)
+        return pybamm.simplify_if_constant(
+            pybamm.Multiplication(other, self), keep_domains=True
+        )
 
     def __matmul__(self, other):
         """return a :class:`MatrixMultiplication` object"""
-        return pybamm.MatrixMultiplication(self, other)
+        return pybamm.simplify_if_constant(
+            pybamm.MatrixMultiplication(self, other), keep_domains=True
+        )
 
     def __rmatmul__(self, other):
         """return a :class:`MatrixMultiplication` object"""
-        return pybamm.MatrixMultiplication(other, self)
+        return pybamm.simplify_if_constant(
+            pybamm.MatrixMultiplication(other, self), keep_domains=True
+        )
 
     def __truediv__(self, other):
         """return a :class:`Division` object"""
-        return pybamm.Division(self, other)
+        return pybamm.simplify_if_constant(
+            pybamm.Division(self, other), keep_domains=True
+        )
 
     def __rtruediv__(self, other):
         """return a :class:`Division` object"""
-        return pybamm.Division(other, self)
+        return pybamm.simplify_if_constant(
+            pybamm.Division(other, self), keep_domains=True
+        )
 
     def __pow__(self, other):
         """return a :class:`Power` object"""
-        return pybamm.Power(self, other)
+        return pybamm.simplify_if_constant(pybamm.Power(self, other), keep_domains=True)
 
     def __rpow__(self, other):
         """return a :class:`Power` object"""
-        return pybamm.Power(other, self)
+        return pybamm.simplify_if_constant(pybamm.Power(other, self), keep_domains=True)
 
     def __lt__(self, other):
         """return a :class:`Heaviside` object"""
-        return pybamm.Heaviside(self, other, equal=False)
+        return pybamm.simplify_if_constant(
+            pybamm.Heaviside(self, other, equal=False), keep_domains=True
+        )
 
     def __le__(self, other):
         """return a :class:`Heaviside` object"""
-        return pybamm.Heaviside(self, other, equal=True)
+        return pybamm.simplify_if_constant(
+            pybamm.Heaviside(self, other, equal=True), keep_domains=True
+        )
 
     def __gt__(self, other):
         """return a :class:`Heaviside` object"""
-        return pybamm.Heaviside(other, self, equal=False)
+        return pybamm.simplify_if_constant(
```

Comment:
Instead of having `pybamm.simplify_if_constant( )` in every function can we have `self = pybamm.simplify_if_constant(self)` in the `__init__` of `Symbol`? Maybe this screws with the tree though?

Reply:
I tried but python doesn't let you assign self in the `__init__`, only its arguments (or at least I couldn't figure out how to), so this seems a bit safer